File: Ejercicio4/filterManager.rb
require_relative 'filterChain'

class FilterManager
  def initialize(target)
    @filter_chain = FilterChain.new(target)
  end

  def add_filter(filter)
    @filter_chain.add_filter(filter)
  end

  def mandar_mensaje(mensaje)
    @filter_chain.ejecutar(mensaje)
  end
end

File: P2/Ejercicio1/lib/filterManager.dart
import 'filterChain.dart';
import 'struct_credenciales.dart';
import 'target.dart';
import 'sensorFiltro.dart';

class FilterManager {
  late FilterChain _filterChain;

  FilterManager(Target target, SensorFiltro sensor) {
    _filterChain = FilterChain(target, sensor);
  }

  void addFilter(dynamic filter) {
    _filterChain.addFilter(filter);
  }

  void mandarMensaje(Credenciales mensaje) {
    _filterChain.ejecutar(mensaje);
  }
}
